src/python/pants/backend/jvm/tasks/coursier_resolve.py

```diff
@@ -535,62 +535,69 @@ def _map_coord_to_resolved_jars(cls, result, coursier_cache_path, pants_jar_path
         {
           "coord": "a",
           "dependencies": ["b", "c"],
-          "files": [ ["", "a.jar"], ["sources", "a-sources.jar"] ]
+          "file": "a.jar"
         },
         {
           "coord": "b",
           "dependencies": [],
-          "files": [ ["", "b.jar"] ]
+          "file": "b.jar"
         },
         {
           "coord": "c",
           "dependencies": [],
-          "files": [ ["", "c.jar"] ]
-        }
+          "file": "c.jar"
+        },
+        {
+          "coord": "a:sources",
+          "dependencies": ["b", "c"],
+          "file": "a-sources.jar"
+        },
       ]
     }
 
     Should return:
     {
-      "a": { ResolvedJar(classifier='', path/cache_path="a.jar"),
-             ResolvedJar(classifier='sources', path/cache_path="a-sources.jar") },
-      "b": { ResolvedJar(classifier='', path/cache_path="b.jar") },
-      "c": { ResolvedJar(classifier='', path/cache_path="c.jar") },
+      "a":               ResolvedJar(classifier='', path/cache_path="a.jar"),
+      "a:sources":       ResolvedJar(classifier='sources', path/cache_path="a-sources.jar"),
+      "b": ResolvedJar(classifier='', path/cache_path="b.jar"),
+      "c": ResolvedJar(classifier='', path/cache_path="c.jar"),
     }
 
     :param result: coursier json output
     :param coursier_cache_path: coursier cache location
     :param pants_jar_path_base: location under pants workdir to store the symlink to the coursier cache
-    :return: a map from org:name:version to a set of resolved jars.
+    :return: a map from maven coordinate to a resolved jar.
     """
 
-    coord_to_resolved_jars = defaultdict(set)
+    coord_to_resolved_jars = dict()
 
     for dep in result['dependencies']:
+      simple_coord = dep['coord']
+      jar_path = dep.get('file', None)
+      if not jar_path:
+        # NB: Not all coordinates will have associated files.
+        #     This is fine. Some coordinates will just have dependencies.
+        continue
```

Comment:
Is there an example about this?

Reply:
Yes. I should add a test though. I don't have an example off the top of my head.